src/groupByKey.ts

```diff
@@ -2,6 +2,43 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+interface Student {
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
+}
+
+interface Book {
+  author: string;
+  country: string;
+  imageLink: string;
+  language: string;
+  pages: number;
+  title: string;
+  year: number;
+}
+
+export function groupByKey(
+  items: [],
+  key: string,
+): GroupsMap<Book[] | Student[]> {
+  const values: string[] = [];
+
+  for (const item of items) {
+    const value = item[key];
```

Comment:
Accessing the property using bracket notation without type checking can lead to runtime errors if the `key` does not exist on the `item`. Consider adding a type guard to ensure that `item` has the property `key`.